source/assets/components/RecipeCard.js

```diff
@@ -0,0 +1,273 @@
+  card.appendChild(recipeTitle);
+
+  //time element
+  const time = document.createElement("time");
+  const totalTime = searchForKey(data, "totalTime");
+  time.textContent = convertTime(totalTime);
+  card.appendChild(time);
+
+  // button element - check recipe
+  const checkButton = document.createElement("button");
+  checkButton.classList.add("cook");
+  checkButton.textContent = "View Recipe  ";
+  card.appendChild(checkButton);
+  checkButton.addEventListener("click", function (event) {
+    showCheck(data);
+  });
+
+   this.shadowRoot.appendChild(styleElem);
+   this.shadowRoot.appendChild(card);
+
+  }
+}
+
+
+/**
+ * Recursively search for a key nested somewhere inside an object
+ * @param {Object} object the object with which you'd like to search
+ * @param {String} key the key that you are looking for in the object
+ * @returns {*} the value of the found key
+ */
+function searchForKey(object, key) {
+  var value;
+  Object.keys(object).some(function (k) {
+    if (k === key) {
+      value = object[k];
+      return true;
+    }
+    if (object[k] && typeof object[k] === 'object') {
+      value = searchForKey(object[k], key);
+      return value !== undefined;
+    }
+  });
+  return value;
+}
+
+/**
+ * Extract the URL from the given recipe schema JSON object
+ * @param {Object} data Raw recipe JSON to find the URL of
+ * @returns {String} If found, it returns the URL as a string, otherwise null
+ */
+function getUrl(data) {
+  if (data.url) return data.url;
+  if (data['@graph']) {
+    for (let i = 0; i < data['@graph'].length; i++) {
+      if (data['@graph'][i]['@type'] == 'Article') return data['@graph'][i]['@id'];
+    }
+  };
+  return null;
+}
+
+/**
+ * Similar to getUrl(), this function extracts the organizations name from the
+ * schema JSON object. It's not in a standard location so this function helps.
+ * @param {Object} data Raw recipe JSON to find the org string of
+ * @returns {String} If found, it retuns the name of the org as a string, otherwise null
+ */
+function getOrganization(data) {
+  if (data.publisher?.name) return data.publisher?.name;
+  if (data['@graph']) {
+    for (let i = 0; i < data['@graph'].length; i++) {
+      if (data['@graph'][i]['@type'] == 'Organization') {
+        return data['@graph'][i].name;
+      }
+    }
+  };
+  return null;
+}
+
+/**
+ * Converts ISO 8061 time strings to regular english time strings.
+ * Not perfect but it works for this lab
+ * @param {String} time time string to format
+ * @return {String} formatted time string
+ */
+function convertTime(time) {
+  let timeStr = '';
+
+  // Remove the 'PT'
+  time = time.slice(2);
+
+  let timeArr = time.split('');
+  if (time.includes('H')) {
+    for (let i = 0; i < timeArr.length; i++) {
+      if (timeArr[i] == 'H') return `${timeStr} hr`;
+      timeStr += timeArr[i];
+    }
+  } else {
+    for (let i = 0; i < timeArr.length; i++) {
+      if (timeArr[i] == 'M') return `${timeStr} min`;
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/cse110-fa21-group16/cse110-fa21-group16/pullRequest?prid=8394473)
```suggestion
      if (timeArr[i] == 'M') {return `${timeStr} min`;}
```